app/presenters/search_result_presenter.rb

```diff
@@ -8,29 +8,80 @@ class SearchResultPresenter
            :es_score,
            to: :search_result
 
-  def initialize(search_result, metadata)
+  def initialize(search_result, metadata, doc_index, doc_count, finder_name, debug_score, highlight)
     @search_result = search_result
     @metadata = metadata
+    @index = doc_index + 1
+    @count = doc_count
+    @finder_name = finder_name
+    @debug_score = debug_score
+    @highlight = highlight
   end
 
   def to_hash
     {
-      title: title,
-      link: link,
-      summary: summary,
-      is_historic: is_historic,
-      government_name: government_name,
-      metadata: metadata,
-      show_metadata: show_metadata,
-      format: format,
-      es_score: es_score,
+      link: {
+        text: title,
+        path: link,
+        description: summary_text,
+        data_attributes: {
+          track_category: "navFinderLinkClicked",
+          track_action: "#{@finder_name}.#{@index}",
+          track_label: link,
+          track_options: {
+            dimension28: @count,
+            dimension29: title
+          }
+        }
+      },
+      metadata: structure_metadata,
+      metadata_raw: metadata,
+      subtext: subtext,
+      highlight: @highlight,
+      highlight_text: highlight_text
     }
   end
 
   def link
     search_result.path
   end
 
+  def structure_metadata
+    component_metadata = {}
+
+    if show_metadata
+      metadata.each do |meta|
+        label = meta[:hide_label] ? "<span class='govuk-visually-hidden'>#{meta[:label]}:</span>" : "#{meta[:label]}:"
+
+        if meta[:is_date]
+          value = "<time datetime='#{meta[:machine_date]}'>#{meta[:human_date]}</time>"
+        else
+          value = meta[:value]
+        end
+
+        component_metadata[meta[:label]] = "#{label} #{value}".html_safe
```

Comment:
Could we use [`sanitize`](https://apidock.com/rails/ActionView/Helpers/SanitizeHelper/sanitize) here, rather than `html_safe`?

Reply:
It's a shame we had to move this out of the ERB templates. Could use ERB in this case? https://ruby-doc.org/stdlib-2.6.3/libdoc/erb/rdoc/ERB.html#class-ERB-label-Introduction

Reply:
If I read the doc for sanitize right, it strips out HTML tags - but I want to preserve them. Is that right?

Reply:
Yeah, moving the logic out of the ERB seemed like a wrench but actually I think the code is easier to read in a presenter and we shouldn't really have so much complex logic in the view.

Reply:
Sanitize will preserve your HTML while helping to prevent against XSS code injection by stripping things that aren't whitelisted (e.g. <script>). In contrast, html_safe won't change the string, it just marks it as safe to render as HTML.

Reply:
I didn't know about this! This is brilliant. Changed, done, thank you.